webapp/src/app/shared/phaser/entities/registry/npc.ts

```diff
@@ -116,6 +121,7 @@ export class NPC extends CCEntity {
 		let x = charSettings.x || 0;
 		let y = charSettings.y || 0;
 		if (charSettings.animSheet) {
+			// noinspection SuspiciousTypeOfGuard
```

Comment:
Why was this added? All builds and linters pass without this line. If  `charSettings.animSheet` can indeed be a string then the types are wrong.

Reply:
Linters pass, but typescript itself complains after updating. 
Problem is the type only matches for the number props not the other and i couldn't figure out how to fix that and still have a somewhat readable code

Reply:
The types are wrong in this case. The input can be a string and that's why it's catched here and filled with the correct object.

Input should probably have another type to reflect this, but the whole npc entity misses a few cases. We should first understand how the game handles all the different cases and then see if this solution still makes sense before refactoring it

Reply:
In that case, adding a `//TODO` should be enough for this PR.